include/boost/geometry/algorithms/detail/is_simple/linear.hpp

```diff
@@ -285,19 +285,17 @@ struct is_simple_multilinestring
     static inline bool apply(MultiLinestring const& multilinestring,
                              Strategy const& strategy)
     {
-        typedef per_linestring<Strategy> per_ls;
-
         // check each of the linestrings for simplicity
         // but do not compute self-intersections yet; these will be
         // computed for the entire multilinestring
-        if ( ! detail::check_iterator_range
-                 <
-                     per_ls, // do not compute self-intersections
-                     true // allow empty multilinestring
-                 >::apply(boost::begin(multilinestring),
-                          boost::end(multilinestring),
-                          per_ls(strategy))
-             )
+        // return true for empty multilinestring
+
+        using per_ls = per_linestring<Strategy>; // do not compute self-intersections
```

Comment:
Here too, it's unclear what is checked below. I know that this is how it was implemented so I'd be ok with leaving it for now as it is but we could think about changing it.

Reply:
I agree, renamed it to `not_simple`.